src/FlightMap/Widgets/QGCInstrumentWidget.qml

```diff
@@ -55,6 +59,111 @@ ColumnLayout {
+            _currentCommand = QGCInstrumentWidget.WinchCommands.RELEASE;
+            winchDeliver.checked = winchRetract.checked = false;
+            if (checked) {slider.visible = true; slider.confirmText = qsTr("Emergency release winch")}
+            else {slider.visible = false}
+        }
+    }
+
+    QGCButton {
+        id:                         winchDeliver
+        backRadius:                 10
+        showBorder:                 true
+        height:                     _outerRadius
+        Layout.fillWidth:           true
+        text:                       "Deliver payload"
+        checkable:                  true
+        Image {
+            id:                     downArrow
+            source:                 "/qmlimages/ArrowDirection.svg"
+            height:                 parent.height
+            width:                  height
+            x:                      10
+            transform: Rotation {
+                origin.x:           downArrow.width/2
+                origin.y:           downArrow.height/2
+                angle:              180
+            }
+        }
+        onClicked: {
+            _currentCommand = QGCInstrumentWidget.WinchCommands.DELIVER;
+            winchEmergency.checked = winchRetract.checked = false;
+            if (checked) {slider.visible = true; slider.confirmText = qsTr("Perform payload drop")}
+            else {slider.visible = false}
+        }
+    }
+
+    QGCButton {
+        id:                         winchRetract
+        backRadius:                 10
+        showBorder:                 true
+        height:                     _outerRadius
+        Layout.fillWidth:           true
+        text:                       "Retract winch"
+        checkable:                  true
+        Image {
+            id:                     upArrow
+            source:                 "/qmlimages/ArrowDirection.svg"
+            height:                 parent.height
+            width:                  height
+            x:                      10
+        }
+        onClicked: {
+            _currentCommand = QGCInstrumentWidget.WinchCommands.RETRACT;
+            winchEmergency.checked = winchDeliver.checked = false;
+            if (checked) {slider.visible = true; slider.confirmText = qsTr("Perform winch retract")}
+            else {slider.visible = false}
+        }
+    }
+
+    SliderSwitch {
+        id:                             slider
+        property var _vehicle:          QGroundControl.multiVehicleManager.activeVehicle
+        visible:                        false
+        confirmText:                    qsTr("Not visible")
+        Layout.fillWidth:               true
+        onAccept: {
+            winchDeliver.checked = winchEmergency.checked = winchRetract.checked = false;
+            _vehicle.sendCommand(_WinchNodeID, _MAVLink_CMD_ID, 1, 1, _currentCommand, 1, 1);
+            _currentCommand = -1;
+            visible = false;
+        }
```

Comment:
Same as above, make a function for clarity.
Also, you do not validation on _currentCommand. What if it is -1? What happens then?